Add unit tests and refactor code

basvandriel opened the pull request:
This PR includes several updates across configuration, initialization, and import structures to enhance the project’s maintainability and code organization. This also adds unit tests to understand the code better and to debug it.

**Config Updates:**
- `.gitignore` extended to exclude temporary directories and settings files specific to the development environment.

**Package Initialization:**
- Added `__init__.py` in app to ensure package recognition by Python, enhancing modularity.

**Code Refactoring:**
- Standardized import statements across modules to use relative paths, improving consistency and potentially simplifying future refactoring efforts.

knrdl commented:
Thanks for your interest in the project. I agree that unit tests are a missing piece for this project.

Your changes are hard to read because of the code reformatting. I've formatted the code like this: https://github.com/knrdl/acme-ca-server/blob/main/.github/workflows/lint.yml#L34
Please adapt this code style for now. Changing it might be a separate PR.

Currently I'm not planning to release this project as a python package. I only use the docker image as distribution method. But I can definitely see the benefits.

Please explain why you switched everything to relative imports. Is there a technical reason? I used absolute imports when importing top level modules and relative imports only for intra-module imports. As stated by PEP8:

> Absolute imports are recommended, as they are usually more readable and tend to be better behaved

basvandriel commented:
Hi 

> Thanks for your interest in the project. I agree that unit tests are a missing piece for this project.
> 
> Your changes are hard to read because of the code reformatting. I've formatted the code like this: https://github.com/knrdl/acme-ca-server/blob/main/.github/workflows/lint.yml#L34 Please adapt this code style for now. Changing it might be a separate PR.
> 
> Currently I'm not planning to release this project as a python package. I only use the docker image as distribution method. But I can definitely see the benefits.
> 
> Please explain why you switched everything to relative imports. Is there a technical reason? I used absolute imports when importing top level modules and relative imports only for intra-module imports. As stated by PEP8:
> 
> > Absolute imports are recommended, as they are usually more readable and tend to be better behaved

Hi @knrdl, I switched back to the old linting configuratio, such as the use of single quotes. I did however format the code using ruff, meaning some line breaks have been applied. Would you like me to revert this process too? 

I switched to relative imports within the `app` directory to make importing work when unit testing directly from `app`. When importing this from another root, the absolute imports did not work as expected. However, one pending thing is that the `run.sh` testing script doesn't yet pass - the relative import are not getting recognized. I'm currently working on that. An idea might be to package the project but not distribute for easy importing?

basvandriel commented:
Instead of navigating into the `app` folder to run `uvicorn`, I kept the working directory the same and imported the module like this: `app.main:app`.

knrdl commented:
Please have a look. I've added a basic pytest integration with CI and pyproject.toml. 
This should work without modifying the application code or Dockerfile.

knrdl commented:
Thanks for your contribution to this project. My goal was to create a solid base for the unit tests. Your work helped therefore a lot. Please feel free to open a new PR if you like to add some more tests. There are still many parts left to be tested.
